x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1344,6 +1346,16 @@ private List<CustomAuthenticator> getCustomAuthenticatorFromExtensions(SecurityE
         }
     }
 
+    private AuthorizedProjectsResolver getCustomAuthorizedProjectsResolverOrDefault(
+        SecurityExtension.SecurityComponents extensionComponents
+    ) {
+        final AuthorizedProjectsResolver customAuthorizedProjectsResolver = findValueFromExtensions(
+            "authorized projects resolver",
+            extension -> extension.getAuthorizedProjectsSupplier(extensionComponents)
```

Comment:
Should we allow only to "internal" extensions to provide `AuthorizedProjectsResolver`? Meaning should we also check `isInternalExtension`?